src/Orleans/Messaging/GatewayProviderFactory.cs

```diff
@@ -13,11 +14,11 @@ internal class GatewayProviderFactory
         private readonly IServiceProvider serviceProvider;
         private readonly Logger logger;
 
-        public GatewayProviderFactory(ClientConfiguration cfg, IServiceProvider serviceProvider)
+        public GatewayProviderFactory(ClientConfiguration cfg, IServiceProvider serviceProvider, ILoggerFactory loggerFactory)
```

Comment:
I see that most classes inject ILoggerFactory and then get the logger from it, instead of injecting `LegacyLogger<T>`. Is it because they are not using DI? because injecting that one would be a lot more preferable where possible

Reply:
ideally we just want to inject loggerFactory into DI, and get ILogger or ILoggerFactory from DI right? why are we injecting LegacyLogger<T>?

Reply:
we are moving away from LegacyLogger. why injectting it will be preferable?

Reply:
because it's the expected usage even when we move to `ILogger<T>`. There we won't inject LoggingFactory just to get the ILogger instance, we rely on DI. Why not with our legacy logger too?